src/main/java/org/springframework/data/elasticsearch/core/query/highlight/HighlightCommonParameters.java

```diff
@@ -15,10 +15,12 @@
  */
 package org.springframework.data.elasticsearch.core.query.highlight;
 
+import co.elastic.clients.elasticsearch._types.query_dsl.Query;
```

Comment:
This would pull Elasticsearch classes into the code outside of the `org.springframework.data.elasticsearch.client.elc` package. We cannot do that. There might be different clients in the future - as there have been in the past, and there are projects depending on Spring Data Elasticsearch (like spring-data-opensearch), they would be forced to use classes from the Elasticsearch client library.

Reply:
So there should be a highlight query wrapper in package `org.springframework.data.elasticsearch.client.elc` like `NativeQuery`, and `HighlightCommonParameters` just depends on this wrapper right?

Reply:
Seems that I misunderstood what you mean before. Things are not that simple and using a wrapper still causes the `HighlightCommonParameters` in spring-data-elsticsearch `core` package depending on classes in elasticsearch-java indirectly.

Maybe:
1. there should be a brand new class (say `A`) for highlight query used by `HighlightCommonParameters` in `core` which has nothing to do with the classes in elasticsearch-java client;
2. a corresponding converter(say `A_Processor`) in `org.springframework.data.elasticsearch.client.elc` will handle the conversion from `A` to `co.elastic.clients.elasticsearch._types.query_dsl.Query` which is used by highlight query class in elasticsearch-java;

Reply:
I think the above `A` should be `org.springframework.data.elasticsearch.core.query.Query`, and there is already a way to convert from  `org.springframework.data.elasticsearch.core.query.Query` to  ` `org.springframework.data.elasticsearch.core.query.Query` in `RequestConverter`.

Reply:
I think the above `A` should be `org.springframework.data.elasticsearch.core.query.Query`, and there is already a way to convert from  `org.springframework.data.elasticsearch.core.query.Query` to  ` `org.springframework.data.elasticsearch.core.query.Query` in `RequestConverter`.

Reply:
You found out by yourself what to change, ne need for me to write that 😁